src/creature_states.c

```diff
@@ -4170,60 +4170,106 @@ TbBool process_creature_hunger(struct Thing *thing)
     return false;
 }
 
+TbBool creature_is_hostile_towards(const struct Thing *fightng, const struct Thing *enmtng)
+{
+    struct CreatureStats* crstat = creature_stats_get_from_thing(fightng);
+    for (int i = 0; i < CREATURE_TYPES_MAX; i++)
+    {
+        if ((crstat->hostile_towards[i] == enmtng->model) || (crstat->hostile_towards[i] == CREATURE_ANY))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+TbBool creature_can_be_hostile(const struct Thing *fightng, const struct Thing *enmtng)
+{
+    // Creatures cannot be hostile towards allies if influenced by CTA.
+    if (creature_affected_by_call_to_arms(fightng) || creature_affected_by_call_to_arms(enmtng))
+    {
+        return false;
+    }
+    // Creatures cannot be hostile towards allies if it's part of a group.
+    if (creature_is_group_member(fightng) || creature_is_group_member(enmtng))
+    {
+        return false;
+    }
+    // Creatures cannot be hostile towards allies if they are working.
+    if ((get_creature_state_type(fightng) == CrStTyp_Work) || (get_creature_state_type(enmtng) == CrStTyp_Work))
+    {
+        return false;
+    }
+    // Lastly, check if both creatures doesn't have hostility set between each other.
```

Comment:
bad grammar again. 

`Lastly, check if neither creature has hostility set towards the other.`